public/services/palantir/palantir_registry.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import EventEmitter from 'events';
+import { Palantir, Palantiri } from '../../types';
+
+export class PalantirRegistry extends EventEmitter {
+  private registry: Palantiri = new Map();
+  private errorOutput: Palantir = {
+    key: 'error',
+    suggestion: 'Error',
+  };
+
+  // TODO: close suggestion
+  public open(item: Palantir) {
+    this.emit('onSuggestion', {
+      suggestion: this.getSuggestion(item.key),
+    });
+  }
+
+  public register(item: Palantir | Palantir[]): void;
+  public register(item: unknown) {
+    if (Array.isArray(item)) {
+      item.forEach((palantir: Palantir) =>
+        this.registry.set(palantir.key, {
+          key: palantir.key,
+          description: palantir.description,
+          suggestion: palantir.suggestion,
+        })
+      );
+    } else {
+      const palantir = item as Palantir;
+      this.registry.set(palantir.key, palantir);
```

Comment:
sorry could you provide an example?

Reply:
Sure. Just wondering if we need to refactor like this:
```suggestion
      this.registry.set(palantir.key, {
          key: palantir.key,
          description: palantir.description,
          suggestion: palantir.suggestion,
        });
```
It would be consistent with above.

Reply:
ah i see what you mean. do you think i should take it one step further and make a helper function to map it it both places?

Reply:
Sure. I think that will be reasonable.